sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_credential.py

```diff
@@ -0,0 +1,37 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+import six
+
+
+class SharedKeyCredential(object):
+    """Credential type used for authenticating the client
+    with a subscription key.
+
+    :param str subscription_key: The subscription key to your Text Analytics
+        or Cognitive Services account.
+    """
+    def __init__(self, subscription_key):
+        if not isinstance(subscription_key, six.string_types):
+            raise TypeError("Please provide the subscription key as a string.")
+        self._subscription_key = subscription_key
+
+    @property
+    def subscription_key(self):
```

Comment:
I believe the guidelines say we shouldn't expose this.

Reply:
Yes, it's listed as a SHOULD NOT, but we've decided to keep it in.

Reply:
Curious why we'd go against the guidelines here?

Reply:
See discussion here: https://github.com/Azure/azure-sdk-for-python/pull/9406#discussion_r367160473